server/src/main/java/io/crate/auth/Credentials.java

```diff
@@ -145,4 +147,23 @@ public Predicate<Role> jwtPropertyMatch() {
         }
         return null;
     }
+
+    /**
+     * Matches on CrateDB username.
+     * Used when 'auth.host_based.jwt.iss' is set.
+     * Can match exactly one user.
+     * @return NULL if no lookup is needed (Basic auth).
+     */
+    @Nullable
+    public Predicate<Role> tokenMatch() {
```

Comment:
Maybe personal preferences, but for me this explains more what the method/returned-predicate does. I'd also prefer renaming `jwtPropertyMatch` to `matchByJwtProperties`.
```suggestion
    public Predicate<Role> matchByTokenUsername() {
```

Or we just use one method `matchByToken(boolean checkJwtProperties)` and switch logic by the boolean. This would avoid spreading the different logic into the code base and thus avoid the `isGlobal ? tokenMatch : jwtPropertiesMatch` 

```suggestion
    public Predicate<Role> matchByToken(boolean checkJwtProperties) {
```

Reply:
Also flipped flag in https://github.com/crate/crate/pull/16763/commits/eebabc20122358636436ab944c0d1800e1d973dc